### ISSUE TITLE
Hide failed/succeeded pods in 'kubectl get pods' by default 

### ISSUE THREAD

**Issue by JanetKuo** [kubernetes/kubernetes#12102]
**Situation**: `kubectl get pods` gives you all pods, including the failed/succeeded ones. 
```
NAME                                           READY     STATUS       RESTARTS   AGE
fluentd-cloud-logging-kubernetes-minion-102n   1/1       Running      0          22d
fluentd-cloud-logging-kubernetes-minion-rkd8   1/1       Running      0          22d
hello-world                                    0/1       ExitCode:0   0          48m
hello-world-2                                  0/1       ExitCode:0   0          40m
```

**Solution**: It'd make more sense for the users if we hide those pods (they may no longer want to check) by default. If the users do want to see all pods, they can use `kubectl get pods -a` or `kubectl get pods --all` instead. This solution aligns with `docker ps -a`. 

Ref #7660

**Comment by JanetKuo** [comment]
cc @lavalamp @derekwaynecarr

**Comment by derekwaynecarr** [comment]
It would also be good if we hide via fieldSelector and not client side.  There are still some pieces of code in kubectl that are filtering client side.

**Comment by JanetKuo** [comment]
Fixed in #12112.
